main.cpp

```diff
@@ -5,110 +5,642 @@ Project 5: Part 1 / 4
+{   
+    std::string planeStatus = "stays on the ground";
+    if (airDirection == "down" && upwardForce)
+    {
+        planeStatus = "goes up";
+    }
+    std::cout << "When the air goes " << airDirection << " the upward force creates lift and the plane " << planeStatus << std::endl;
+}
+
+bool PlaneWings::reduceDrag (float tailwind)
+{
+    if(tailwind < 10.f) 
+    return true;
+    return false;
+}
+void PlaneWings::lowersLandingSpeed (int drag, bool landed, float landingSpeed )
+{
+    if(!landed)
+    {
+    landingSpeed -= drag;
+    }
+}
 
 /*
- new UDT 4:
- with 2 member functions
- */
+new UDT 4:
+with 2 member functions
+*/
+struct Kitchen 
+{
+    Range range;
+    Range::RangeControls rangeControls;
+    
+    Kitchen();
+    ~Kitchen();
+    
+    void designSpace(std::string addSlightVariation, int wallWidth);
+    void safetyAlertViaWiFi(int currentOvenTemp, int setOvenTemp, int currentTime);
+};
+
+Kitchen::Kitchen() 
+{
+    std::cout << "Kitchen being constructed." << std::endl;
+}
+
+Kitchen::~Kitchen() 
+{
+    std::cout << "Kitchen being destructed." << std::endl;
+}
+
+void Kitchen::designSpace(std::string addSlightVariation, int wallWidth) 
+{
+    std::string themeColor =
+    addSlightVariation + rangeControls.controlPanelColor;
+    int measureSpaceForFridge = wallWidth - range.width;
+    
+    std::cout << "Kitchen will be designed to go with " << themeColor
+    << std::endl;
+    std::cout << "We still have enough space for a " << measureSpaceForFridge
+    << " wide fridge." << std::endl;
+}
+
+void Kitchen::safetyAlertViaWiFi(int currentOvenTemp, int setOvenTemp, int currentTime) 
+{
+    if (rangeControls.supportsWifi)
+    {
+        if (currentOvenTemp > 0) 
+        {
+            while (currentTime < 2400 && currentOvenTemp < setOvenTemp) 
+            {
+                currentOvenTemp += 10;
+                int mins;                
+                for (mins = 1; mins <= 10; ++mins) 
+                {
+                    std::cout << "Alert to oven user : oven is on at " << currentOvenTemp << std::endl;
+                }
+            }
+            std::cout << "Auto turn off the oven ." << std::endl;
+        }
+        std::cout << "Oven is not on." << std::endl;
+    }
+    std::cout << "This model doesn't support WiFi feature." << std::endl;
+}
 
 /*
- new UDT 5:
- with 2 member functions
- */
+new UDT 5:
+with 2 member functions
+*/
+struct Concorde 
+{
+    PlaneWings pw;
+    
+    Concorde();
+    ~Concorde();
+    
+    int lengthDeIcingPanel (int variation);
+    int topSpeed(int cNumEngines);
+};
+
+Concorde::Concorde() 
+{
+    std::cout << "Concorde being constructed." << std::endl;
+}
+
+Concorde::~Concorde() 
+{
+    std::cout << "Concorde being destructed." << std::endl;
+}
+
+int Concorde::lengthDeIcingPanel (int variation)
+{
+    int n = 1;
+    int hypotenuse;
+    int cWingSpan = pw.wingSpan  / 2 - variation;
+    return hypotenuse = cWingSpan / n;
+}
```

Comment:
never write `return a = b;`
What value are you returning?   Are you returning `a`?  are you returning `b`?
Write clear code that doesn't suffer from ambiguities like these.
```
a = b / c;
return a;
```
`n` is 1.  What is the point of dividing by `1`? 
initialize `hypotenuse` with a value. 
- [ ] revise